yt_dlp/extractor/common.py

```diff
@@ -2499,7 +2499,7 @@ def _extract_mpd_formats(self, *args, **kwargs):
 
     def _extract_mpd_formats_and_subtitles(
             self, mpd_url, video_id, mpd_id=None, note=None, errnote=None,
-            fatal=True, data=None, headers={}, query={}):
+            fatal=True, data=None, headers={}, query={}, multi_period=False):
```

Comment:
I don't like how the entire format of the return is changed by this param - the function name is even wrong when using this, which is confusing and will become unmaintainable. If this is truly desired, we would need to make a new function like `_extract_mpd` and then keep `_extract_mpd_formats_and_subtitles` just as a thin wrapper around it.

But returning merged fragment list like your previous commit and fixing up with ffmpeg is a cleaner solution imo. See my comment on discord for how to do that.

Reply:
> But returning merged fragment list like your previous commit and fixing up with ffmpeg is a cleaner solution imo. See my comment on discord for how to do that.

I agree with this, also downloading a certain format is way easier with this approach

Reply:
Comment from pukkadan on discord:
> In that case, we want to add a new private key in the formats dict (maybe is_dash_periods?) and change the fixup condition to 
`(info_dict.get('is_live') or info_dict.get('is_dash_periods')) and downloader == 'DashSegmentsFD'`

File: `yt_dlp/YoutubeDL.py`